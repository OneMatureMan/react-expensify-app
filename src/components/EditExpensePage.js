import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {

    onRemoveExpense = () => {
        this.props.startRemoveExpense({id: this.props.expense.id})
        this.props.history.push('/dashboard')
    }

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id,expense)
        this.props.history.push('/dashboard')
    }

    render() {
        return  (
            <div>
                <h1>Edit Expense</h1>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit= {this.onSubmit}
                />
                <button onClick={this.onRemoveExpense}>
                    Remove Expense
                </button>
            </div>
        )

    }   
} 

const mapDispatchToProps = (dispatch, props) => ({
    startRemoveExpense : (id) => dispatch(startRemoveExpense(id)),
    startEditExpense: (id,expense) => dispatch(startEditExpense(id,expense))
})

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);