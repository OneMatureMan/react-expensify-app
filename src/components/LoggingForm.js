import React from 'react';
import { Link } from 'react-router-dom';


class LoggingForm extends React.Component {
    state = {
        email: '',
        password: '',
        error: ''
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }))
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }))
    }


    render() {
        return (
            <form 
                onSubmit={async (e) => {
                    e.preventDefault()
                    const error = await this.props.signingFunction(this.state.email,this.state.password);
                    if (error) {
                        this.setState(() => ({ error }));
                    }
                }}
                className="logging-form"
            >
                {this.state.error && <p className='form__error'>{this.state.error}</p>}
                <label className='logging-form__label'>Email address</label>
                <input className='logging-form__input' type="email" onChange={this.onEmailChange}/>
                <label className='logging-form__label'>Password</label>
                <input className='logging-form__input' type="password" onChange={this.onPasswordChange}/>
                <button 
                    className={`button ${this.props.styledButton}`} 
                >   
                    {this.props.buttonName}
                </button>
            </form>  
        )
    }
}


export default LoggingForm;