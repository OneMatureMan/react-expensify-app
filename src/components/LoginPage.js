import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';


export const LoginPage = ({ startLogin }) => {
    const onSubmit = (e) => {
        e.preventDefault();
        startLogin()
    }
    return (
        <form onSubmit={onSubmit}>
            <label>Username: </label>
            <input type="text" name="username"/>
            <label>Password: </label>
            <input type="password" name="password"/>
            <button>Login</button>
        </form>
    )
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);
