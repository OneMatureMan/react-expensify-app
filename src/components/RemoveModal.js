import React from 'react';
import Modal from 'react-modal';


const RemoveModal = (props) => (
    <Modal
        isOpen={props.isOpen}
        contentLabel="Selected Option"
        onRequestClose={props.handleCloseModal}
        ariaHideApp={false}
        closeTimeoutMS={200}
        className='modal'
    >
        <h3 className='modal__title'>Remove Expense?</h3>
        {/* <p className='modal__body'>Are you sure that you want to remove the expense?</p> */}
        <button className='button button--remove' onClick={props.onRemoveExpense}>Remove</button>
        <button className='button' onClick={props.handleCloseModal}>Close</button>
    </Modal>
)


export default RemoveModal;