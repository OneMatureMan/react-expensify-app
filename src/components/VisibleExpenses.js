import React from 'react';




const VisibleExpenses = ({ textFilter, startDate, endDate, allExpensesCount, expensesCount}) => {
    const expenseWord = (allExpensesCount - expensesCount) === 1 ? 'expense' : 'expenses';

    return (
        ((textFilter === '') && (startDate === null) && (endDate === null) || (allExpensesCount === expensesCount)) ? (
            <p className='visible-expenses-message'>Showing all expenses, no filters applied</p>
        ) : (
            <p className='visible-expenses-message'>Not showing <span>{allExpensesCount - expensesCount}</span> {expenseWord}</p>
        )
    )
    
}

export default VisibleExpenses;