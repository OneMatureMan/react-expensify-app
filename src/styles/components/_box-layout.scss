.box-layout {
    background: url('/images/bg.jpg');
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100vh;
    width: 100vw;
    background-size: cover;
}

.box-layout__box {
    border-radius: 3px;
    background: fade-out(white, 0.15);
    text-align: center;
    padding: $l-size $m-size;
    width: 35rem;

}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}