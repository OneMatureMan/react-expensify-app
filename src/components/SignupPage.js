import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { startSignupWithEmail } from '../actions/auth';
import LoggingForm from './LoggingForm';


class SignupPage extends React.Component {
    render() {
        return (
            <div className='box-layout'>
                <div className="box-layout__box">
                    <Link to='/' className='navlink-wrapper'>
                        <button className='button button--navlink'>Back</button>
                    </Link>
                    <h1 className="box-layout__title">Sign up</h1>
                    <LoggingForm
                        styledButton={'button--signup'}
                        buttonName={'Sign up'}
                        signingFunction={this.props.startSignupWithEmail}
                    />
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startSignupWithEmail: (email, password) => dispatch(startSignupWithEmail(email, password))
})


export default connect(undefined,mapDispatchToProps)(SignupPage);