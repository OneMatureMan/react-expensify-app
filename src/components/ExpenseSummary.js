import React from 'react';
import { connect } from 'react-redux'
import { Link} from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import VisibleExpenses from './VisibleExpenses';


export const ExpenseSummary = ({expensesCount, expensesTotal, allExpensesCount, textFilter, startDate, endDate}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral(expensesTotal/100).format('$0,0.00')

    return (
        <div className='page-header'>
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedTotal}</span></h1>
                <VisibleExpenses
                    allExpensesCount={allExpensesCount}
                    expensesCount={expensesCount}
                    textFilter={textFilter}
                    startDate={startDate}
                    endDate={endDate}
                />
                <div className="page-header__actions">
                    <Link to='/create'>
                        <button className='button'>Add Expense</button>
                    </Link>
                </div>
            </div>
        </div>
    )
}
    


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses,state.filters);
    const expensesTotal = selectExpensesTotal(visibleExpenses);
    
    return {
        allExpensesCount: state.expenses.length,
        expensesCount: visibleExpenses.length,
        expensesTotal,
        textFilter: state.filters.text,
        startDate: state.filters.startDate,
        endDate: state.filters.endDate
    }
};


export default connect(mapStateToProps)(ExpenseSummary);