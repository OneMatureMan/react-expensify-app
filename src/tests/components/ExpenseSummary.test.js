import React from 'react';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';



test('Should return 1 expense with a total of 1200', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={1200} />)
    expect(wrapper).toMatchSnapshot()
});

test('Should return 2 expenses with a total of 80000', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={80000} />)
    expect(wrapper).toMatchSnapshot()
});