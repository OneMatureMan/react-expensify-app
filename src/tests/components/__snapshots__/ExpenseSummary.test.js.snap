// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return 1 expense with a total of 1200 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $12.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        to="/create"
      >
        <button
          className="button"
        >
          Add Expense
        </button>
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should return 2 expenses with a total of 80000 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $800.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        to="/create"
      >
        <button
          className="button"
        >
          Add Expense
        </button>
      </Link>
    </div>
  </div>
</div>
`;
